pkg/operator/configobservation/images/observe_images.go

```diff
@@ -1,42 +1,111 @@
 package images
 
 import (
+	"bytes"
+	"encoding/json"
+
 	"github.com/golang/glog"
 
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 
-	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/configobservation"
 	"github.com/openshift/library-go/pkg/operator/configobserver"
+
+	"github.com/openshift/cluster-openshift-apiserver-operator/pkg/operator/configobservation"
 )
 
-func ObserveInternalRegistryHostname(genericListers configobserver.Listers, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
+func ObserveImageConfig(genericListers configobserver.Listers, existingConfig map[string]interface{}) (map[string]interface{}, []error) {
 	listers := genericListers.(configobservation.Listers)
 	var errs []error
 	prevObservedConfig := map[string]interface{}{}
 
+	// first observe all the existing config values so that if we get any errors
+	// we can at least return those.
 	internalRegistryHostnamePath := []string{"imagePolicyConfig", "internalRegistryHostname"}
 	if currentInternalRegistryHostname, _, _ := unstructured.NestedString(existingConfig, internalRegistryHostnamePath...); len(currentInternalRegistryHostname) > 0 {
-		unstructured.SetNestedField(prevObservedConfig, currentInternalRegistryHostname, internalRegistryHostnamePath...)
+		err := unstructured.SetNestedField(prevObservedConfig, currentInternalRegistryHostname, internalRegistryHostnamePath...)
+		if err != nil {
+			return prevObservedConfig, append(errs, err)
+		}
+	}
+
+	externalRegistryHostnamePath := []string{"imagePolicyConfig", "externalRegistryHostnames"}
+	o, _, err := unstructured.NestedSlice(existingConfig, externalRegistryHostnamePath...)
+	err = unstructured.SetNestedSlice(prevObservedConfig, o, externalRegistryHostnamePath...)
+	if err != nil {
+		return prevObservedConfig, append(errs, err)
+	}
+
+	allowedRegistriesForImportPath := []string{"imagePolicyConfig", "allowedRegistriesForImport"}
+	o, _, err = unstructured.NestedSlice(existingConfig, allowedRegistriesForImportPath...)
+	err = unstructured.SetNestedSlice(prevObservedConfig, o, allowedRegistriesForImportPath...)
```

Comment:
if block?

Reply:
this (as above) was intentional... if the existing config has an empty slice for this value, i want to set the prevObservedConfig to be that empty slice, so that the config gets updated with that empty slice value instead of whatever slice content may be there currently.